### PR TITLE
FE_tag_test

### PR DESCRIPTION
**Pull request by Artanis7488**
I have added comprehensive unit tests for the WTag component to ensure its correct functionality. The tests cover proper redirection to pages as specified and the verification of style variations when isActive is true or false.

### CODE REVIEW

File: frontend/src/components/atoms/Tag/tag.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { render, fireEvent } from '@testing-library/react'
+import WTag from './tag'
+import { AllInclusive } from '@mui/icons-material'
+import './index.scss'
+
+describe('WTag', () => {
+    it('Should redirect to pages correctly as specified', () => {
+        const path = '/ruta-especifica'
+        const loggedMessages = []
```
Comment (RicseDev) [inline frontend/src/components/atoms/Tag/tag.test.tsx:9]: why arguments ? 
no especified arguments  =>         const loggedMessages: any[] = []

File: frontend/src/components/atoms/Tag/tag.test.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        //Restore the original console.log function
+        console.log = originalLog
+
+        //Verify that it has been called with the expected message
+        expect(loggedMessages).toContain(`Ir a la ruta ${path}`);
+    })
+
+    it('Should display style variations when isActive is true or false', () => {
+        const { container } = render(<WTag icon={AllInclusive} text="TagLink" isActive={true} />);
+        expect(container.firstChild).toHaveClass('tagLink--active');
```
Comment (RicseDev) [inline frontend/src/components/atoms/Tag/tag.test.tsx:29]: add to check if an active is really being created in the tag

example 
```jsx
        expect(container.firstChild).toHaveClass('tagLink--active');
        expect(container.firstChild).toMatchSnapshot();
```

```jsx

        expect(containerFalse.firstChild).not.toHaveClass('tagLink--active');
        expect(containerFalse.firstChild).toMatchSnapshot();
```
use snapchot  
with snapchot toamr screenshot and place it in your preview of your pr as an argument of a your test generated that

File: frontend/src/components/atoms/Tag/tag.test.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        const originalLog = console.log
+        console.log = (...args) => {
+            loggedMessages.push(args.join(' '))
+        };
+
+        const { getByText } = render(<WTag icon={AllInclusive} text={path} path={path} />); //Chage "text" to "path"
+        fireEvent.click(getByText(path)); //Change "TagLink" to "path"
+
+        //Restore the original console.log function
+        console.log = originalLog
```
Comment (RicseDev) [inline frontend/src/components/atoms/Tag/tag.test.tsx:21]: delete extra comment 
replace comment
use an important-only comment format //TODO : comments

**Review by RicseDev — APPROVED** [review]

**Review by RicseDev — COMMENTED** [review]
correct the proposed comments

File: frontend/src/components/atoms/Tag/tag.test.tsx (hunk 1 above)
Comment (Artanis7488) [reply frontend/src/components/atoms/Tag/tag.test.tsx:9]: I've corrected that.

File: frontend/src/components/atoms/Tag/tag.test.tsx (hunk 2 above)
Comment (Artanis7488) [reply frontend/src/components/atoms/Tag/tag.test.tsx:29]: You're right. I've already implemented the snapshots.

File: frontend/src/components/atoms/Tag/tag.test.tsx (hunk 3 above)
Comment (Artanis7488) [reply frontend/src/components/atoms/Tag/tag.test.tsx:21]: I've already removed it.

**Comment by Artanis7488** [comment]
image
image
Three snapshots were created to test the WTag component.
The first two snapshots show style variations with "isActive" as true or false.
The third snapshot checks if the component redirects to a specific page, confirming its proper functionality.

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "interactive"
```
Comment (AnthonyRicse98) [inline .vscode/settings.json:2]: delete file

File: frontend/package.json [hunk 5]
```diff
@@ -49,6 +49,7 @@
     "identity-obj-proxy": "^3.0.0",
     "jest": "^29.7.0",
     "jest-environment-jsdom": "^29.7.0",
+    "jest-snapshot": "^29.7.0",
     "jest-transform-stub": "^2.0.0",
```
Comment (AnthonyRicse98) [inline frontend/package.json:53]: dont modified

**Review by RicseDev — APPROVED** [review]

**Review by AnthonyRicse98 — APPROVED** [review]

**Pull request merged**